src/main/kotlin/com/tylerthrailkill/helpers/prettyprint/PrettyPrint.kt

```diff
@@ -91,11 +105,11 @@ private class PrettyPrinter(val tabSize: Int, val writeTo: Appendable, val wrapp
      * indentation level of any closing bracket.
      */
     private fun <T> Iterable<T>.ppContents(currentDepth: String, separator: String = "", f: (T) -> Unit) {
-        val list = this.toMutableList()
+        val list = this.toList()
 
         if (!list.isEmpty()) {
-            f(list.removeAt(0))
-            list.forEach {
+            f(list.first())
+            list.drop(1).forEach {
```

Comment:
interesting change. Does this just remove the need for a mutable list?

Reply:
Yes. The more mutable things that can be avoided, the less chance that effects can somehow work together to create bugs